lib/ts/recipe/thirdparty/index.ts

```diff
@@ -0,0 +1,143 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { getNormalisedUserContext } from "../../utils";
+import { UserType } from "../authRecipeWithEmailVerification/types";
+import { RecipeFunctionOptions } from "../emailpassword";
+import Recipe from "./recipe";
+import { InputType, StateObject, PreAndPostAPIHookAction, PreAPIHookContext, PostAPIHookContext } from "./types";
+
+export default class Wrapper {
+    static init(config: InputType) {
+        return Recipe.init(config);
+    }
+
+    static getOAuthState(input?: { userContext?: any }): {
+        status: "OK";
+        state: StateObject | undefined;
+    } {
+        const recipeInstance = Recipe.getInstanceOrThrow();
+
+        return recipeInstance.recipeImplementation.getOAuthState({
+            userContext: getNormalisedUserContext(input?.userContext),
+            config: recipeInstance.config,
+        });
+    }
+
+    static setOAuthState(input: { state: StateObject; userContext?: any }): {
+        status: "OK";
+    } {
+        const recipeInstance = Recipe.getInstanceOrThrow();
+
+        return recipeInstance.recipeImplementation.setOAuthState({
+            ...input,
+            userContext: getNormalisedUserContext(input.userContext),
+            config: recipeInstance.config,
+        });
+    }
+
+    static getThirdPartyLoginRedirectURLWithQueryParams(input: {
+        thirdPartyProviderId: string;
+        thirdPartyRedirectionURL: string;
+        state?: StateObject;
+        userContext?: any;
+    }): Promise<
+        | {
+              status: "ERROR";
+          }
+        | {
+              status: "OK";
+              url: string;
+          }
+    > {
+        const recipeInstance = Recipe.getInstanceOrThrow();
+
+        return recipeInstance.recipeImplementation.getThirdPartyLoginRedirectURLWithQueryParams({
+            ...input,
+            config: recipeInstance.config,
+            userContext: getNormalisedUserContext(input.userContext),
+        });
+    }
+
+    static getOAuthAuthorisationURL(input: {
+        thirdPartyProviderId: string;
+        userContext?: any;
+        options?: RecipeFunctionOptions;
+    }): Promise<{
+        status: "OK";
+        url: string;
+        fetchResponse: Response;
+    }> {
+        const recipeInstance = Recipe.getInstanceOrThrow();
+
+        return recipeInstance.recipeImplementation.getOAuthAuthorisationURL({
+            ...input,
+            config: recipeInstance.config,
+            userContext: getNormalisedUserContext(input.userContext),
+        });
+    }
+
+    static signInAndUp(input: {
+        thirdPartyProviderId: string;
+        thirdPartyRedirectionURL: string;
+        thirdPartyProviderClientId?: string;
```

Comment:
rename to remove thirParty prefix.

Reply:
We also need to provide an optional `authCode` input (just like we do in email verification token function)

Reply:
Done. If no `authCode` is provided and there is no `code` in the query params the function throws an error